### ISSUE TITLE
Option allowed-redefined-builtins has no effect

### ISSUE THREAD

**Issue by sattlerc** [pylint-dev/pylint#10160]
### Bug description

The option allowed-redefined-builtins appears to have no effect.

Here is an example file `a.py`:

```python
id = 1
```

### Command used

```shell
pylint --allowed-redefined-builtins id a.py
```

### Pylint output

```
************* Module a
a.py:2:0: C0305: Trailing newlines (trailing-newlines)
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
a.py:1:0: W0622: Redefining built-in 'id' (redefined-builtin)
a.py:1:0: C0103: Constant name "id" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

The message `a.py:1:0: W0622: Redefining built-in 'id' (redefined-builtin)` should not appear.

### Pylint version

```shell
pylint 3.3.3
astroid 3.3.8
Python 3.13.1 (main, Dec  4 2024, 18:05:56) [GCC 14.2.1 20240910]
```

**Comment by zenlyj** [comment]
thanks for the report. based on the discussions in #3643, it seems that this behavior is intentional. the decision was made to exclude module level re-definitions from the `allowed-redefined-builtins` option.

```
id = 1  # warning raised
```
```
def foo():
    id = 1  # no warning raised
```

in my opinion, this behavior should be documented as I can see how it could be misleading to users.

**Comment by zenlyj** [comment]
another case where the option has no effect:
```
def foo():
    global id
    id = 1  # [redefined-builtin]
```
